backend/core/database.py

```diff
@@ -1038,18 +1056,43 @@ def handle_delete_hijack(self, message):
                 redis_hijack_key = redis_key(
                     raw["prefix"], raw["hijack_as"], raw["type"]
                 )
+                redis_hijack = self.redis.get(redis_hijack_key)
                 if self.redis.sismember("persistent-keys", raw["key"]):
                     purge_redis_eph_pers_keys(self.redis, redis_hijack_key, raw["key"])
 
-                self.wo_db.execute("DELETE FROM hijacks WHERE key=%s;", (raw["key"],))
-                self.wo_db.execute(
-                    "DELETE FROM bgp_updates WHERE %s = ANY(hijack_key) AND array_length(hijack_key,1) = 1;",
-                    (raw["key"],),
-                )
-                self.wo_db.execute(
-                    "UPDATE bgp_updates SET hijack_key = array_remove(hijack_key, %s);",
-                    (raw["key"],),
+                log.debug(
+                    "redis-entry for {}: {}".format(redis_hijack_key, redis_hijack)
                 )
+                self.wo_db.execute("DELETE FROM hijacks WHERE key=%s;", (raw["key"],))
+                if (
+                    redis_hijack
+                    and len(yaml.safe_load(redis_hijack).get("bgpupdate_keys", [])) > 0
+                ):
+                    log.debug("deleting hijack using cache for bgp updates")
+                    redis_hijack = yaml.safe_load(redis_hijack)
+                    log.debug(
+                        "bgpupdate_keys {} for {}".format(
+                            redis_hijack["bgpupdate_keys"], redis_hijack_key
+                        )
+                    )
+                    self.wo_db.execute(
+                        "DELETE FROM bgp_updates WHERE %s = ANY(hijack_key) AND handled = true AND array_length(hijack_key,1) = 1 AND key = ANY(%s);",
+                        (raw["key"], list(redis_hijack["bgpupdate_keys"])),
+                    )
+                    self.wo_db.execute(
+                        "UPDATE bgp_updates SET hijack_key = array_remove(hijack_key, %s) WHERE handled = true AND key = ANY(%s);",
+                        (raw["key"], list(redis_hijack["bgpupdate_keys"])),
+                    )
+                else:
```

Comment:
this is fallback in case redis does not contain any related info, correct? So it is the old way of doing things with a cache-miss?

Reply:
if the hijack is resolved or not active then it's not in redis